recipes/fftw/all/patches/fix-simd.patch

```diff
@@ -0,0 +1,42 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -134,6 +134,7 @@ include (CheckCCompilerFlag)
+ if (ENABLE_SSE)
+   foreach (FLAG "-msse" "/arch:SSE")
+     unset (HAVE_SSE CACHE)
++	unset (HAVE_SSE)
```

Comment:
Could you explain this patch? What does it change in the context of conan? Seems useless, but maybe I'm wrong.

Reply:
The patch doesn't fix anything in terms of conan itself, but it does fix the part of the build I'm trying to enable with this PR. At least on my machine, I have to unset the variable both with `CACHE` and without or otherwise it doesn't test the second flag, and so fails to do anything on Windows.

Reply:
It's weird, is there an issue opened upstream?

I'll test it on my machine when I have time.

Reply:
It is now. I opened it myself after detecting the problem while making this PR.

The minimum you need to test it on a CMakeLists.txt:
```cmake
include (CheckCCompilerFlag)
foreach (FLAG "-msse2" "/arch:SSE2")
  unset (HAVE_SSE2 CACHE)
  #unset(HAVE_SSE2)
  check_c_compiler_flag (${FLAG} HAVE_SSE2)
  if (HAVE_SSE2)
    set (SSE2_FLAG ${FLAG})
    break()
  endif ()
endforeach ()
```
And you'll notice it only tests once (the first flag), then fails, of course, because it didn't test for the MSVC version of the flag. At least on my machine it does it. Only unsetting twice it works.

Reply:
You can probably reproduce it on a *NIX machine just by inverting the order of the flags, so it fails first.

Reply:
They merged [my PR](https://github.com/FFTW/fftw3/pull/232) upstream, so the patch should be a fix just up to the next version.

Reply:
As it is a PR related to build and accepted by the upstream, it can be accepted.

Reply:
:+1:

Reply:
May I ask you to add the PR in the header of the patch?